### PR TITLE
[CBRD-23729] Enhance compactdb utility

### PR DESCRIPTION
**Pull request by mhoh3963**
(1) To execute compactdb with specified classes on SA mode, I modified codes for using -i option and class arguments on compactdb.
    If use -i or class arguments, modified compactdb will be skipped phase3 (reclaim catalog, deleted files, file_tracker) on SA mode.

    example)
        cubrid compactdb -S -v -i compact.txt demodb
        cubrid compactdb -S -v demodb athlete event record

(2) To execute compactdb on C/S mode at standby server, I added option such as --standby which is hidden option.

    example)
        cubrid compactdb -C -v -i compact.txt --standby demodb@standby
        cubrid compactdb -C --standby -v demodb@standby athlete event record

### CODE REVIEW

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/executables/compactdb.c (40 incorrect indentation(s) found)
```
   126       compact_usage (arg->argv0);
   133 	{
   134 	  PRINT_AND_LOG_ERR_MSG (msgcat_message
   135 				 (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_COMPACTDB, COMPACTDB_MSG_FAILURE));
   136 	  return ER_GENERIC_ERROR;
   137 	}
   140 	{
   141 	  tables[i - 1] = utility_get_option_string_value (arg_map, OPTION_STRING_TABLE, i);
   142 	}
   156       status = compactdb_start (verbose_flag, input_filename, tables, table_size - 1);
   186   bool skip_phase3 = false;
   195 	{
   196 	  return ER_FAILED;
   197 	}
   200 	{
   201 	  status = get_class_mops (input_class_names, input_class_length, &class_mops, &num_class_mops);
   202 	  if (status != NO_ERROR)
   203 	    {
   205 	    }
   206 	  skip_phase3 = true;
```
... More than 20 lines are omitted.
#### File: src/executables/compactdb_cl.c (1 incorrect indentation(s) found)
```
   883   else
```
#### File: src/executables/util_admin.c (1 incorrect indentation(s) found)
```
   567   {COMPACT_STANDBY_CS_MODE_L, 0, 0, COMPACT_STANDBY_CS_MODE_S},
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

File: src/executables/compactdb.c [hunk 1]
```diff
@@ -102,15 +104,43 @@ compactdb (UTIL_FUNCTION_ARG * arg)
...
+
+  table_size = utility_get_option_string_table_size (arg_map);
+  if (table_size > 1 && input_filename != NULL)
     {
       compact_usage (arg->argv0);
       return ER_GENERIC_ERROR;
     }
+  else if (table_size > 1)
+    {
+      tables = (char **) malloc (sizeof (char *) * table_size - 1);
```
Comment (kisoo-han) [inline src/executables/compactdb.c:131]: tables = (char **) malloc (sizeof (char *) * **(table_size - 1)**);

File: src/executables/compactdb_common.c [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+  char *ptr = NULL;
+
+  if (input_filename == NULL || class_list == NULL || num_class_mops == NULL)
+    {
+      return ER_FAILED;
+    }
+
+  *class_list = NULL;
+  *num_class_mops = 0;
+  if (get_num_requested_class (input_filename, &num_class) != NO_ERROR)
```
Comment (kisoo-han) [inline src/executables/compactdb_common.c:184]: Suggestion:
`if (get_num_requested_class (input_filename, &num_class) != NO_ERROR || num_class == 0)`
And remove next if-statement.

File: src/executables/compactdb_common.c [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+      class_names[i] = NULL;
+    }
+
+  for (i = 0; i < num_class; ++i)
+    {
+      if (fgets ((char *) buffer, DB_MAX_IDENTIFIER_LENGTH, input_file) == NULL)
+	{
+	  status = ER_FAILED;
+	  goto end;
+	}
```
Comment (kisoo-han) [inline src/executables/compactdb_common.c:217]: _indentation style (for all following if-statements)_:
```
if (fgets ((char *) buffer, DB_MAX_IDENTIFIER_LENGTH, input_file) == NULL)
  {
     status = ER_FAILED;
  }
```

File: src/executables/compactdb_common.c [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+    }
+
+  if (class_names)
+    {
+      for (i = 0; i < num_class; i++)
+	{
+	  if (class_names[i] != NULL)
+	    {
+	      free (class_names[i]);
+	      class_names[i] = NULL;
```
Comment (kisoo-han) [inline src/executables/compactdb_common.c:262]: Suggestion: use of **free_and_init ()  macro** for two statements
`free_and_init (class_names[i]);`

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/executables/compactdb_common.c (18 incorrect indentation(s) found)
```
   120 	{
   122 	}
   128 	{
   131 	}
   133 	{
   138 	}
   209 	{
   212 	}
   216 	{
   218 	}
   220 	{
   222 	}
   225 	{
   228 	}
   232 	{
   235 	}
   253 	{
   255 	}
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/executables/compactdb_common.c (6 incorrect indentation(s) found)
```
   120 	{
   122 	}
   128 	{
   131 	}
   133 	{
   138 	}
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/executables/compactdb_common.c (11 incorrect indentation(s) found)
```
   120 	{
   121 	  goto error;
   122 	}
   128 	{
   129 	  (*class_list)[(*num_class_list)] = class_;
   130 	  (*num_class_list)++;
   131 	}
   133 	{
   134 	  printf (msgcat_message (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_COMPACTDB, COMPACTDB_MSG_CLASS),
   137 	  printf (msgcat_message (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_COMPACTDB, COMPACTDB_MSG_INVALID_CLASS));
   138 	}
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

**Comment by mhoh3963** [comment]
thank you for your review. I applied your opinion to code.

File: src/executables/compactdb_common.c (hunk 3 above)
Comment (mhoh3963) [reply src/executables/compactdb_common.c:212]: Indentation is changed  by executing "indent" command.
Why do you know to change indentation ?
Please let me know solution if you know.

Comment (kisoo-han) [reply src/executables/compactdb_common.c:212]: Sorry, I have no idea on it. Anyway, you changed indentation as I requested.

**Review by kisoo-han — APPROVED** [review]

**Pull request merged**